Source/Core/Core/HW/DVD/AMMediaboard.cpp

```diff
@@ -0,0 +1,1829 @@
+
+  for (u32 i = 0; i < length; i += 0x10)
+  {
+    INFO_LOG_FMT(AMMEDIABOARD, "GC-AM: {:08x} {:08x} {:08x} {:08x}", memory.Read_U32(address + i),
+                 memory.Read_U32(address + i + 4), memory.Read_U32(address + i + 8),
+                 memory.Read_U32(address + i + 12));
+  }
+}
+
+void FirmwareMap(bool on)
+{
+  s_firmware_map = on;
+}
+
+void InitKeys(u32 key_a, u32 key_b, u32 key_c)
+{
+  s_GCAM_key_a = key_a;
+  s_GCAM_key_b = key_b;
+  s_GCAM_key_c = key_c;
+}
+
+static File::IOFile OpenOrCreateFile(const std::string& filename)
+{
+  // Try opening for read/write first
+  if (File::Exists(filename))
+    return File::IOFile(filename, "rb+");
+
+  // Create new file
+  return File::IOFile(filename, "wb+");
+}
+
+void Init()
+{
+  std::ranges::fill(s_media_buffer_32, 0);
+  std::ranges::fill(s_network_buffer, 0);
+  std::ranges::fill(s_network_command_buffer, 0);
+  std::ranges::fill(s_firmware, -1);
+  std::ranges::fill(s_sockets, SOCKET_ERROR);
+  std::ranges::fill(s_allnet_buffer, 0);
+  std::ranges::fill(s_allnet_settings, 0);
+
+  s_firmware_map = false;
+  s_test_menu = false;
+
+  s_last_error = SSC_SUCCESS;
+
+  s_GCAM_key_a = 0;
+  s_GCAM_key_b = 0;
+  s_GCAM_key_c = 0;
+
+  const std::string base_path = File::GetUserPath(D_TRIUSER_IDX);
+
+  s_netcfg = OpenOrCreateFile(base_path + "trinetcfg.bin");
+  s_netctrl = OpenOrCreateFile(base_path + "trinetctrl.bin");
+  s_extra = OpenOrCreateFile(base_path + "triextra.bin");
+  s_dimm = OpenOrCreateFile(base_path + "tridimm_" + SConfig::GetInstance().GetGameID() + ".bin");
+  s_backup = OpenOrCreateFile(base_path + "backup_" + SConfig::GetInstance().GetGameID() + ".bin");
+
+  if (!s_netcfg.IsOpen())
+    PanicAlertFmt("Failed to open/create: {}", base_path + "trinetcfg.bin");
+  if (!s_netctrl.IsOpen())
+    PanicAlertFmt("Failed to open/create: {}", base_path + "trinetctrl.bin");
+  if (!s_extra.IsOpen())
+    PanicAlertFmt("Failed to open/create: {}", base_path + "s_extra.bin");
+  if (!s_dimm.IsOpen())
+    PanicAlertFmt("Failed to open/create: {}", base_path + "s_dimm.bin");
+  if (!s_backup.IsOpen())
+    PanicAlertFmt("Failed to open/create: {}", base_path + "s_backup.bin");
+
+  // This is the firmware for the Triforce
+  const std::string sega_boot_filename = base_path + "segaboot.gcm";
+
+  if (!File::Exists(sega_boot_filename))
+  {
+    PanicAlertFmt("Failed to open segaboot.gcm({}), which is required for test menus.",
+                  sega_boot_filename.c_str());
+    return;
+  }
+
+  File::IOFile sega_boot(sega_boot_filename, "rb+");
+  if (!sega_boot.IsOpen())
+  {
+    PanicAlertFmt("Failed to read: {}", sega_boot_filename);
+    return;
+  }
+
+  const u64 length = std::min<u64>(sega_boot.GetSize(), sizeof(s_firmware));
+  sega_boot.ReadBytes(s_firmware, length);
+
+  s_test_menu = true;
+}
+
+u8* InitDIMM(u32 size)
+{
+  if (size == 0)
+    return nullptr;
+
+  if (!s_dimm_disc)
+  {
+    s_dimm_disc = new (std::nothrow) u8[size];
```

Comment:
Unless I'm mistaken, we can get rid of this naked new by using `Common::UniqueBuffer<u8>`.  Then, `new[size]` can be replaced with `.reset(size)`; and `delete[]` with `.clear()`.